tests/test_models.py

```diff
@@ -585,6 +593,43 @@ def test_jsondict(self):
         self.assertEqual(m1.data, {})
         self.assertRaises(ValueError, MyData, data='NonDict')
 
+    def test_urltype(self):
+        m1 = MyUrlModel(
+            url=u"https://example.com", url_all_scheme=u"magnet://example.com",
+            url_custom_scheme=u"ftp://example.com"
+            )
+        self.session.add(m1)
+        self.session.commit()
+        self.assertEqual(m1.url, u"https://example.com")
+        self.assertEqual(m1.url_all_scheme, u"magnet://example.com")
+        self.assertEqual(m1.url_custom_scheme, u"ftp://example.com")
+
+    def test_urltype_invalid(self):
+        with self.assertRaises(StatementError):
+            m1 = MyUrlModel(url=u"example.com")
+            self.session.add(m1)
+            self.session.commit()
```

Comment:
Is this right? Do we allow urls without scheme?

Reply:
Add more parameters: `relative_scheme` and `relative_path`, both `False` by default (if `relative_path` is `True`, it turns on `relative_scheme` as well).